### ISSUE TITLE
Swift 4.1 error: Ambiguous use of operator '=='

### ISSUE THREAD

**Issue by richardtop** [google/promises#27]
Hi, 
I got an error when working with _Swift 4.1_ and _Xcode 9.3 beta 3 (9Q117m)_, when installing the framework using _CocoaPods v1.4.0_:
![image](https://user-images.githubusercontent.com/8013017/36679180-43023a06-1b1b-11e8-9917-9e3384d434c1.png)

Not critical now, stable _Xcode 9.2_ and _Swift 4_ work correctly.

Removing this piece of code makes it compile:
```Swift
    static func == (lhs: When<Value?>, rhs: When<Value?>) -> Bool {
      switch (lhs, rhs) {
      case (.value(let lhs), .value(let rhs)):
        switch (lhs, rhs) {
        case (nil, nil):
          return true
        case (nil, _?), (_?, nil):
          return false
        case let (lhs?, rhs?):
          return lhs == rhs
        }
      case (.error(let lhs), .error(let rhs)):
        return (lhs as NSError).isEqual(rhs as NSError)
      case (.value, .error), (.error, .value):
        return false
      }
    }
```

Introduction of the [conditional conformance](https://swift.org/blog/conditional-conformance/) in Swift 4.1 might be the root cause.

**Comment by shoumikhin** [comment]
Thanks, Richard!
The issue should be resolved in [1.0.6](https://github.com/google/promises/releases/tag/1.0.6).

**Comment by richardtop** [comment]
Thanks for the quick release!
